### PR TITLE
[WIP][CI:DOCS]Add DevContainers Documentation

### PR DESCRIPTION
**Pull request by jakecorrenti**
Added documentation for setting up DevContainers in Visual Studio Code using Podman instead of Docker.

Adapted from the work done in #15254

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by jakecorrenti** [comment]
Basic use cases are reproducible. However, I need to iron out some edge cases with more complex use cases.

File: docs/tutorials/vscode_devcontainers_tutorial.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+	"workspaceMount": "source=${localWorkspaceFolder},target=/workspace,type=bind,Z",
+	"workspaceFolder": "/workspace",
+
+	"runArgs": ["--userns=keep-id"],
+	"containerUser": "vscode"
+}
+```
+
+`Dockerfile`
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/vscode_devcontainers_tutorial.md:113]: I'm assuming it has to be a Dockerfile vs a Containerfile

**Comment by TomSweeneyRedHat** [comment]
Just a quick late day review, and it looks really good overall.  It would be awesome of you could walk through this in a Gmeet, save the recording, then give me a ptr to it so I could put it up on the YouTube channel.

**Comment by jakecorrenti** [comment]
I could definitely put that together for you

**Comment by fulldecent** [comment]
I followed along on a M1 MacBook Pro. I kept the uid=1000 part and did not know if I needed to change that.

Also I needed to add a `podman machine start`.

End result is this error:


Screenshot 2023-05-24 at 19 30 49

File: docs/tutorials/vscode_devcontainers_tutorial.md (hunk 1 above)
Comment (jakecorrenti) [reply docs/tutorials/vscode_devcontainers_tutorial.md:120]: confirmed that Containerfile does work

**Comment by fulldecent** [comment]
I see the instructions are updated. I followed them, instead using my actual user name instead of $USER in the env variable.

Now got further to this error message:

```
[62455 ms] Start: Run in container: mkdir -p '/root/.vscode-server/bin' && ln -snf '/vscode/vscode-server/bin/linux-arm64/b3e4e68a0bc097f0ae7907b217c1119af9e03435' '/root/.vscode-server/bin/b3e4e68a0bc097f0ae7907b217c1119af9e03435'
[62473 ms] 
[62473 ms] mkdir: cannot create directory ‘/root’: Permission denied
[62473 ms] Exit code 1
[62475 ms] Command in container failed: mkdir -p '/root/.vscode-server/bin' && ln -snf '/vscode/vscode-server/bin/linux-arm64/b3e4e68a0bc097f0ae7907b217c1119af9e03435' '/root/.vscode-server/bin/b3e4e68a0bc097f0ae7907b217c1119af9e03435'
[62475 ms] mkdir: cannot create directory ‘/root’: Permission denied
[62475 ms] Exit code 1
```

**Comment by jakecorrenti** [comment]
@fulldecent what Image did you select? And what does your `devcontainer.json` file look like

**Comment by fulldecent** [comment]
@jakecorrenti I used the Ubuntu image.

And here is the file

```javascript
// Documentation:
//  - https://github.com/devcontainers/images/tree/main/src/jekyll
//  - https://github.com/devcontainers/features/tree/main/src/node
//  - https://github.com/devcontainers/features/tree/main/src/php

// Issue: how to remove the github.copilot dependency?
// https://stackoverflow.com/questions/76037936/how-to-load-github-copilot-without-specifying-in-devcontainer-json
{
    "name": "Jekyll",
    "image": "mcr.microsoft.com/devcontainers/jekyll",
    "features": {
      // Install latest stable Node (we use for validation scripts after build)
      "ghcr.io/devcontainers/features/node:1": {
        "version": "lts"
      }
    },
    "customizations": {
      "vscode": {
        "extensions": ["github.copilot", "html-validate.vscode-html-validate"]
      }
    }
}
```

**Comment by jakecorrenti** [comment]
> @jakecorrenti I used the Ubuntu image.
> 
> And here is the file
> 
> ```js
> // Documentation:
> //  - https://github.com/devcontainers/images/tree/main/src/jekyll
> //  - https://github.com/devcontainers/features/tree/main/src/node
> //  - https://github.com/devcontainers/features/tree/main/src/php
> 
> // Issue: how to remove the github.copilot dependency?
> // https://stackoverflow.com/questions/76037936/how-to-load-github-copilot-without-specifying-in-devcontainer-json
> {
>     "name": "Jekyll",
>     "image": "mcr.microsoft.com/devcontainers/jekyll",
>     "features": {
>       // Install latest stable Node (we use for validation scripts after build)
>       "ghcr.io/devcontainers/features/node:1": {
>         "version": "lts"
>       }
>     },
>     "customizations": {
>       "vscode": {
>         "extensions": ["github.copilot", "html-validate.vscode-html-validate"]
>       }
>     }
> }
> ```

Using the "features" functionality is known to lead to bugs (https://github.com/containers/podman/issues/17313). Investigating.

**Comment by vrothberg** [comment]
Closing the PR due to inactivity.  Feel free to reopen.

**Pull request closed without merging**